application.go

```diff
@@ -1,35 +1,31 @@
-package client
+package sdk
 
 import (
 	"context"
 
 	"github.com/cosmos/gogoproto/grpc"
 	"github.com/pokt-network/poktroll/x/application/types"
-
-	"github.com/pokt-network/shannon-sdk/sdk"
 )
 
-var _ sdk.ApplicationClient = (*applicationClient)(nil)
-
-// applicationClient is an ApplicationClient implementation that uses the gRPC
+// ApplicationClient is an ApplicationClient implementation that uses the gRPC
 // query client of the application module.
-type applicationClient struct {
-	queryClient types.QueryClient
+type ApplicationClient struct {
+	types.QueryClient
 }
 
 // NewApplicationClient creates a new application client with the provided gRPC connection.
-func NewApplicationClient(grpcConn grpc.ClientConn) (sdk.ApplicationClient, error) {
-	return &applicationClient{
-		queryClient: types.NewQueryClient(grpcConn),
-	}, nil
+func NewApplicationClient(grpcConn grpc.ClientConn) *ApplicationClient {
+	return &ApplicationClient{
+		QueryClient: types.NewQueryClient(grpcConn),
+	}
 }
 
 // GetAllApplications returns all applications in the network.
-func (ac *applicationClient) GetAllApplications(
+func (ac *ApplicationClient) GetAllApplications(
 	ctx context.Context,
 ) ([]types.Application, error) {
 	req := &types.QueryAllApplicationsRequest{}
-	res, err := ac.queryClient.AllApplications(ctx, req)
+	res, err := ac.QueryClient.AllApplications(ctx, req)
```

Comment:
Why is `QueryClient` exposed?

Reply:
`QueryClient` needs to become an interface (eventually) as it is being consumed here. Making it public is the first step. SDK should allow passing any struct that implements its needs as the query client (e.g. this will make unit testing significantly easier).

Reply:
#PUC (Please Update Comment)

Reply:
Thanks, updated the PR.